documentation/library_guide/kernel_templates/esimd/radix_sort_by_key.rst

```diff
@@ -0,0 +1,361 @@
+         sycl::host_accessor k_out_acc{keys_out, sycl::read_only};
+         for(std::size_t i = 0; i < n; ++i)
+               std::cout << k_out_acc[i] << ' ';
+         std::cout << '\n';
+
+         sycl::host_accessor v_out_acc{values_out, sycl::read_only};
+         for(std::size_t i = 0; i < n; ++i)
+               std::cout << v_out_acc[i] << ' ';
+         std::cout << '\n';
+      }
+
+      return 0;
+   }
+
+**Output:**
+
+.. code:: none
+
+   3 2 1 5 3 3
+   r o s d t e
+
+   1 2 3 3 3 5
+   s o r t e d
+
+
+.. _radix-sort-by-key-memory-requirements:
+
+-------------------
+Memory Requirements
+-------------------
+
+The algorithm uses global and local device memory (see `SYCL 2020 Specification
+<https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_sycl_device_memory_model>`_)
+for intermediate data storage. For the algorithm to operate correctly, there must be enough memory
+on the device; otherwise, the behavior is undefined. The amount of memory that is required
+depends on input data and configuration parameters, as described below.
+
+Global Memory Requirements
+--------------------------
+
+Global memory is used for copying the input sequence(s) and storing internal data such as radix value counters.
+The used amount depends on many parameters; below is an upper bound approximation:
+
+   N\ :sub:`keys` + N\ :sub:`values` + C * N\ :sub:`keys`
+
+where the sequence with keys takes N\ :sub:`keys` space, the sequence with values takes N\ :sub:`values` space,
+and the additional space is C * N\ :sub:`keys`.
+
+The value of `C` depends on ``param.data_per_workitem``, ``param.workgroup_size``, and ``RadixBits``.
+For ``param.data_per_workitem`` set to `32`, ``param.workgroup_size`` to `64`, and ``RadixBits`` to `8`,
+`C` approximately equals to `1`.
+Incrementing ``RadixBits`` increases `C` up to twice, while doubling either
+``param.data_per_workitem`` or ``param.workgroup_size`` leads to a halving of `C`.
+
+..
+   The estimation above is not very precise and it seems it is not necessary for the global memory.
+   The C coefficient base is actually 0.53 instead of 1.
+   An increment of RadixBits multiplies C by the factor of ~1.5 on average.
+
+   Additionally, C exceeds 1 for radix_sort_by_key,
+   when N is small and the global histogram takes more space than the sequences.
+   This space is small, single WG implementation will be added, therefore this is neglected.
+
+Local Memory Requirements
+-------------------------
+
+Local memory is used for reordering key-value pairs within a work-group,
+and for storing internal data such as radix value counters.
+The used amount depends on many parameters; below is an upper bound approximation:
+
+   N\ :sub:`keys_per_workgroup` + N\ :sub:`values_per_workgroup` + C
+
+where N\ :sub:`keys_per_workgroup` and N\ :sub:`values_per_workgroup` are the amounts of memory
+to store keys and values, respectively.  `C` is some additional space for storing internal data.
```

Comment:
Please remove the double space between "...respectively.  'C' is some..."

Reply:
Thanks, fixed.